isisd/isis_cli.c

```diff
@@ -70,16 +70,42 @@ DEFPY_YANG_NOSH(router_isis, router_isis_cmd,
 	return ret;
 }
 
+struct if_iter {
+	struct vty *vty;
+	const char *tag;
+};
+
+static int if_iter_cb(const struct lyd_node *dnode, void *arg)
+{
+	struct if_iter *iter = arg;
+	const char *tag;
+
+	if (!yang_dnode_exists(dnode, "frr-isisd:isis/area-tag"))
+		return YANG_ITER_CONTINUE;
+
+	tag = yang_dnode_get_string(dnode, "frr-isisd:isis/area-tag");
+	if (strmatch(tag, iter->tag)) {
+		char xpath[XPATH_MAXLEN];
+		const char *name = yang_dnode_get_string(dnode, "name");
+		const char *vrf = yang_dnode_get_string(dnode, "vrf");
+
+		snprintf(
+			xpath, XPATH_MAXLEN,
+			"/frr-interface:lib/interface[name='%s'][vrf='%s']/frr-isisd:isis",
+			name, vrf);
+		nb_cli_enqueue_change(iter->vty, xpath, NB_OP_DESTROY, NULL);
+	}
+
+	return YANG_ITER_CONTINUE;
```

Comment:
Nitpick:
Is it possible to have more than one `router isis WORD` with same name? Otherwise it would be better to stop iterating.

I guess since this iteration is per VRF it should be fine.

Reply:
We're iterating through interfaces here. There may be multiple interfaces using the same router and we're deleting them all, so we shouldn't stop the iteration.

Actually, there's no more dependency between interface/router configs so this code is not needed at all. We can preserve the interface configs even when the router config is deleted. But there may be people relying on this behavior. What do you think?

Reply:
> We're iterating through interfaces here. There may be multiple interfaces using the same router and we're deleting them all, so we shouldn't stop the iteration.

I missed it that it was in the interface XPath, sorry.

> Actually, there's no more dependency between interface/router configs so this code is not needed at all. We can preserve the interface configs even when the router config is deleted. But there may be people relying on this behavior. What do you think?

I agree with you, if `router isis WORD` is removed we should not touch interface configuration. I think there is no harm doing this and there are no blockers anymore.

> But there may be people relying on this behavior.

Just to clarify: the current behavior is if you have the following configuration:

```
interface foo
 ip address 1.2.3.4/5
 ip router isis bar
!
router isis bar
 ! config
!
```

If you call `no router isis bar` you'd get:

```
interface foo
 ip address 1.2.3.4/5
!
```

Which seems inconsistent because if I remember correctly you can configure `ip router isis bar` in interface before `router isis bar` but the removal works the other way around.

Reply:
Yes, you could configure `ip router isis bar` on an interface, but it automatically created the `router isis bar` config, because there was a strict dependency.

Now that this dependency is removed, it is not necessary to do that, but I still preserved the automatic creation of router config, because people can rely on this behavior - they only configure the interface and everything starts working.

Anyway, let's keep the code for now so that this PR only fixes CLI things without changing behavior. We can change it later in a separate PR.